src/constants.js

```diff
@@ -0,0 +1,52 @@
+const charValue = {
+  A:1, B:3, C:3, D:2, E:1, F:4, G:2, H:4, I:1, J:8, K:5, L:1, M:3, N:1, O:1, P:3, Q:10, R:1, S:1, T:1, U:1, V:4, W:4, X:8, Y:4, Z:10
+}
+
+const charFactor = {
+  current: 1,
+  valid: true,
+}
+
+const resetCharFactor = () => {
+  charFactor.current = 1
+  charFactor.valid = true
+  delete charFactor.previous
+}
+
+const getValue = (char) => (!!char.match(/[^A-Za-z]/) ? 0 : charValue[char.toUpperCase()])
+
+const charFactorChangeTrigger = (char) => {
+  if (!!char.match(/[\[\]\{\}]/)) {
```

Comment:
I recommend using regexes sparingly, in this case you could have skipped this by using the `default` statement in your switchcase or by `return`ing instead of `break`ing

Reply:
Thank you for the feedback, is this because it's not very intuitive to read with the escaping or for another reason?

Reply:
Readability is definitely a reason, but mostly I'd say it's just safer. It's very easy to make mistakes while writing regexes and they can also expose your application to security issues (like [redos](https://en.wikipedia.org/wiki/ReDoS))